src/cli/Vaults.ts

```diff
@@ -47,6 +77,50 @@ function makeAddVaultCommand() {
     );
 }
 
+function makeCloneVaultCommand() {
+  return new commander.Command('clone')
+    .description('clone a vault from a peer')
+    .option('--node-path <nodePath>', 'node path')
+    .requiredOption('-p, --public-key <publicKey>', 'public key file path of the peer who has the vault')
+    .requiredOption('-n, --vault-name <vaultName>', 'name of the vault to be cloned')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const nodePath = determineNodePath(options);
+        const publicKeyPath = options.publicKey;
+        const vaultName = options.vaultName;
+
+        // read in public key
+        const publicKey = fs.readFileSync(publicKeyPath);
+
+        const successful = await client.cloneVault(nodePath, vaultName, publicKey.toString());
+        pkLogger(`vault '${vaultName}' cloned ${successful ? 'un-' : ''}successfully`, PKMessageType.SUCCESS);
+      }),
+    );
+}
+
+function makePullVaultCommand() {
+  return new commander.Command('pull')
+    .description('pull a vault from a peer')
+    .option('--node-path <nodePath>', 'node path')
+    .requiredOption('-p, --public-key <publicKey>', 'public key file path of the peer who has the vault')
+    .requiredOption('-n, --vault-name <vaultName>', 'name of the vault to be cloned')
+    .action(
+      actionRunner(async (options) => {
+        const client = PolykeyAgent.connectToAgent();
+        const nodePath = determineNodePath(options);
+        const publicKeyPath = options.publicKey;
+        const vaultName = options.vaultName;
+
+        // read in public key
+        const publicKey = fs.readFileSync(publicKeyPath);
```

Comment:
For `pk` commands you can definitely use `Sync` versions of fs calls. But beware of using this anywhere else especially for the agent.

Although we can standardize and always use the promise versions and never worry about wrong uses of `Sync` versions.